apps/web/components/v2/eventtype/EventLimitsTab.tsx

```diff
@@ -114,13 +114,99 @@ export const EventLimitsTab = (props: Pick<EventTypeSetupInfered, "eventType">)
+          <Controller
+            name="minimumBookingNotice"
+            control={formMethods.control}
+            render={() => {
+              const durationArray = [eventType.minimumBookingNoticeType];
+              const minBookingValue = formMethods.watch("minimumBookingNotice");
+              let newMinBookingValue: number;
+              const convertToNewDurationType = function (
+                prevType: string,
+                newType: string,
+                prevValue: number
+              ) {
+                if (!prevType) {
+                  prevType = eventType.minimumBookingNoticeType;
+                }
+                if (newType == "minute") {
+                  if (prevType == "hour") {
+                    newMinBookingValue = prevValue * 60;
+                  }
+                  if (prevType == "day") {
+                    newMinBookingValue = prevValue * 1440;
+                  }
+                } else if (newType == "hour") {
+                  if (prevType == "minute") {
+                    newMinBookingValue = prevValue / 60;
+                  }
+                  if (prevType == "day") {
+                    newMinBookingValue = prevValue * 24;
+                  }
+                } else if (newType == "day") {
+                  if (prevType == "minute") {
+                    newMinBookingValue = prevValue / 1440;
+                  }
+                  if (prevType == "hour") {
+                    newMinBookingValue = prevValue / 24;
+                  }
+                }
+              };
+
+              const durationTypeOptions = [
+                {
+                  label: t("minutes"),
+                  value: "minute",
+                },
+                {
+                  label: t("hours"),
+                  value: "hour",
+                },
+                {
+                  label: t("days"),
+                  value: "day",
+                },
+              ];
+
+              convertToNewDurationType(
+                "minute",
+                eventType.minimumBookingNoticeType,
+                eventType.minimumBookingNotice
+              );
+              eventType.minimumBookingNotice = newMinBookingValue;
+
+              return (
+                <>
+                  <DurationField
+                    required
+                    label={t("minimum_booking_notice")}
+                    type="number"
+                    placeholder="120"
+                    className="mr-2 w-full"
+                    defaultValue={eventType.minimumBookingNotice}
+                    {...formMethods.register("minimumBookingNotice", { valueAsNumber: true })}
+                  />
+                  <Select
+                    isSearchable={false}
+                    className="mb-2 ml-2 w-1/3"
+                    defaultValue={durationTypeOptions.find(
+                      (option) => option.value === eventType.minimumBookingNoticeType
```

Comment:
If we only say we allow whole numbers in the duration field (eg whole minutes, whole hours or whole days), I think we could make the `noticeType` a client side only select. That would make this change a lot smaller since we don't need that additional field in the DB. 

When we load this page we check the number of minutes, then check whether it rounds to a full amount of days, if not, we do the same for hours, if not, we show minutes. 

What do you think of this idea? I think it makes the solution quite a bit easier to use throughout the rest of our app.

Reply:
@JeroenReumkens I'll add this asap!